app/routes/settings/email-preferences.js

```diff
@@ -8,5 +8,10 @@ export default Route.extend({
   },
   model() {
     return this.get('authManager.currentUser').query('emailNotifications', { include: 'event' });
+  },
+  actions: {
+    willTransition() {
+      this.get('authManager.currentUser').rollbackAttributes();
```

Comment:
Oh yea. My bad. Forgot about that. Then just roll back that specific preference when the save request fails